.github/workflows/ci.yml

```diff
@@ -0,0 +1,172 @@
+    runs-on: ubuntu-latest
+    if: |
+      (github.event_name != 'pull_request_target' ||
+        (github.event.pull_request.head.repo.full_name != 'KaTeX/KaTeX' &&
+          contains(github.event.pull_request.labels.*.name, 'test screenshots'))) &&
+      !contains(toJSON(github.event.commits.*.message), '[skip ci]') &&
+      !contains(toJSON(github.event.commits.*.message), '[ci skip]')
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.result }}
+    
+    steps:
+    - id: set-matrix
+      uses: actions/github-script@v2
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        script: |
+          const SELENIUM_BROWSERS = ["chrome", "firefox"];
+          const BROWSERSTACK_BROWSERS = [{
+            browserName: "safari",
+            browser_version: "13.1",
+            os: "OS X",
+            os_version: "Catalina",
+          }];
+
+          const include = [];
+
+          if (context.eventName !== "pull_request_target") {
+            include.push(...SELENIUM_BROWSERS.map(b => ({
+              browser: b,
+              services: {selenium: {
+                image: `selenium/standalone-${b}:3.141.59-20200525`,
+                ports: ["4444:4444"],
+              }},
+            })));
+          }
+
+          // check access to Browserstack crendential secrets
+          if (context.eventName !== "pull_request" ||
+              context.payload.pull_request.head.repo.full_name === "KaTeX/KaTeX") {
+            if (context.eventName === "pull_request_target") {
+              github.issues.removeLabel({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                name: "test screenshots",
+              });
+            }
+            include.push(...BROWSERSTACK_BROWSERS.map(b => ({
+              browser: b.browserName,
+              services: {},
+              browserstack: b,
+            })));
+          }
+
+          return {browser: include.map(b => b.browser), include};
+
+  screenshotter:
+    runs-on: ubuntu-latest
+    needs: screenshotter_dispatcher
+    strategy:
+      matrix: ${{ fromJson(needs.screenshotter_dispatcher.outputs.matrix) }}
+      fail-fast: false
+    services: ${{ matrix.services }}
+
+    steps:
+    - uses: actions/checkout@v2
+      if: github.event_name != 'pull_request_target'
+      with:
+        submodules: recursive
+    - uses: actions/checkout@v2
+      if: github.event_name == 'pull_request_target'
+      with:
+        ref: refs/pull/${{ github.event.pull_request.number }}/head
+        submodules: recursive
+
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          .yarn/cache
+          .pnp.js
+        key: yarn-deps-v1-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          yarn-deps-v1-
```

Comment:
It's too bad there isn't a way to share steps between jobs.

Reply:
@kevinbarabash I think it'll be added soon. GitHub is already working on a PR: https://github.com/actions/runner/issues/646.